components/states/shared/BackButton.jsx

```diff
@@ -0,0 +1,33 @@
+import React from "react";
+import { ArrowBack } from "@mui/icons-material";
+import { MyContext } from "../../ReservationController";
+import { useActor } from "@xstate/react";
+
+const BackButton = (props) => {
+  const machine = React.useContext(MyContext);
+  const [state, send] = useActor(machine);
+
+  const statesWithoutBackButton = ["reservation", "complete"];
```

Comment:
you don't need to do this is you were to just use `prevState` and not an array, you will don't need to hardcode this, and the way to do it is you just check if its set?